lib/libspl/include/umem.h

```diff
@@ -137,6 +137,15 @@ umem_free(const void *ptr, size_t size __maybe_unused)
 	free((void *)ptr);
 }
 
+/*
+ * umem_free_aligned was readded for supporting portability
+ * with non-POSIX platforms that require a different free
+ * to be used with aligned allocations.
+ */
+#ifndef _WIN32
+#define	umem_free_aligned umem_free
```

Comment:
After seeing the revision, I feel like we should define the _WIN32 case if we are going to have the ifdef at this stage, but it is a minor thing that could be left as is for now.

Reply:
Oh that's great, I was trying not to include any windows specific code since I thought that might get in the way of getting this merged.

Reply:
nit: declaring this as a `static inline` function would be a nice way to make it clear how it should be called.  Something like:

```c
static inline void
umem_free_aligned(void *buf, size_t size __unused)
{
       free(buf);
}
```

We can also simply omit the `#ifndef _WIN32` wrapper until we bring in the windows implementation of this.